### PR TITLE
Optimized bayer to RGB565

### PR DESCRIPTION
**Pull request by bitbank2**
Improved the conversion speed by minimizing memory accesses and combining operations.

### CODE REVIEW

File: src/omv/img/imlib.c [hunk 1]
```diff
@@ -420,6 +420,7 @@ void imlib_bayer_to_rgb565(image_t *img, int w, int h, int xoffs, int yoffs, uin
 {
     int r, g, b;
     for (int y=yoffs; y<yoffs+h; y++) {
+    if (xoffs & 1 || y == 0 || y >= yoffs+h-1) {
```
Comment (kwagyeman) [inline src/omv/img/imlib.c:423]: Fix indentation.

File: src/omv/img/imlib.c [hunk 2]
```diff
@@ -471,8 +472,66 @@ void imlib_bayer_to_rgb565(image_t *img, int w, int h, int xoffs, int yoffs, uin
             g = IM_G826(g);
             b = IM_B825(b);
             *rgbbuf++ = IM_RGB565(r, g, b);
-        }
-    }
+        } // for x
+    } else { // faster way
+        uint32_t l0,l1,l2; // 3 lines we're working with
```
Comment (kwagyeman) [inline src/omv/img/imlib.c:477]: Can you add some comments on what this code is trying to do? I'm having a very hard time following it.

File: src/omv/img/imlib.c (hunk 1 above)
Comment (kwagyeman) [reply src/omv/img/imlib.c:423]: Also, maybe some block comments on what this code is trying to do.

File: src/omv/img/imlib.c (hunk 2 above)
Comment (bitbank2) [reply src/omv/img/imlib.c:508]: added

File: src/omv/img/imlib.c (hunk 1 above)
Comment (bitbank2) [reply src/omv/img/imlib.c:425]: added

**Comment by kwagyeman** [comment]
Please use BAYER->YUV for improving the jpeg compress speed.

**Pull request merged**